src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -85,6 +85,16 @@ const TopicFragment = ({
 }) => {
   const [activeSections, setActiveSections] = React.useState([]);
 
+  React.useEffect(() => {
+    if (searchText.length === 0) {
+      setActiveSections([]);
+    } else if (searchText.length >= 0 && followedTopic.length > 0) {
+      setActiveSections([0]);
+    } else {
+      setActiveSections([]);
+    }
+  }, [searchText, followedTopic]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in the `useEffect` hook seems to be a bit redundant. If `searchText.length === 0`, you're setting `activeSections` to an empty array, and you're doing the same thing if neither of the conditions are met. You could simplify this by using an else statement instead of specifying the condition again.

```diff
-    if (searchText.length === 0) {
-      setActiveSections([]);
-    } else if (searchText.length >= 0 && followedTopic.length > 0) {
-      setActiveSections([0]);
-    } else {
-      setActiveSections([]);
-    }
+    if (searchText.length > 0 && followedTopic.length > 0) {
+      setActiveSections([0]);
+    } else {
+      setActiveSections([]);
+    }
```
This way, you're only checking for the condition where `searchText.length > 0 && followedTopic.length > 0` once, and in all other cases, you're setting `activeSections` to an empty array. This reduces redundancy and makes the code easier to read and maintain.


<!-- This is an auto-generated comment by OSS CodeRabbit -->